src/test/java/com/hubspot/dropwizard/guicier/HK2LinkerTest.java

```diff
@@ -55,4 +73,31 @@ public void explicitGuiceBindingsAreBridgedToHk2() throws ServletException {
         assertThat(resource).isNotNull();
         assertThat(resource.getDAO()).isNotNull();
     }
+
+    @Test
+    public void contextBindingsAreBridgedToGuice() {
+        // This list comes from https://stackoverflow.com/a/35868654
+        List<Class<?>> classes = Arrays.asList(
+            Application.class,
+            javax.ws.rs.core.Configuration.class,
+            ContainerRequestContext.class,
+            HttpHeaders.class,
+            HttpServletRequest.class,
+            HttpServletResponse.class,
+            Providers.class,
+            Request.class,
+            ResourceContext.class,
+            SecurityContext.class,
+            ServletConfig.class,
+            ServletContext.class,
+            UriInfo.class,
+            // Jersey-specific
+            ExtendedUriInfo.class);
+        for (Class<?> clazz : classes) {
+            Binding binding = injector.getExistingBinding(Key.get(clazz));
```

Comment:
Not sure how the test scaffolding works, but could we also call `getInstance` to make sure there's no exception thrown?

Reply:
I tried that, but lots of these fail complaining about circular resolution. They work if I put them inside a resource method and hit the injector, so I can add that as a separate test.